exchanges/btcmarkets/btcmarkets_websocket.go

```diff
@@ -388,58 +376,68 @@ func (b *BTCMarkets) Subscribe(subs []subscription.Subscription) error {
 		signTime := strconv.FormatInt(time.Now().UnixMilli(), 10)
 		strToSign := "/users/self/subscribe" + "\n" + signTime
 		var tempSign []byte
-		tempSign, err = crypto.GetHMAC(crypto.HashSHA512,
-			[]byte(strToSign),
-			[]byte(creds.Secret))
+		tempSign, err = crypto.GetHMAC(crypto.HashSHA512, []byte(strToSign), []byte(creds.Secret))
 		if err != nil {
 			return err
 		}
 		sign := crypto.Base64Encode(tempSign)
-		payload.Key = creds.Key
-		payload.Signature = sign
-		payload.Timestamp = signTime
+		baseReq.Key = creds.Key
+		baseReq.Signature = sign
+		baseReq.Timestamp = signTime
 	}
 
-	if err := b.Websocket.Conn.SendJSONMessage(payload); err != nil {
-		return err
+	var errs error
+	for i, s := range subs {
+		if i == 1 {
+			baseReq.MessageType = addSubscription
+			baseReq.ClientType = clientType
+		}
+
+		req := baseReq
+
+		req.Channels = []string{s.Channel}
+		req.MarketIDs = s.Pairs.Strings()
+
+		err := b.Websocket.Conn.SendJSONMessage(req)
+		if err == nil {
+			err = b.Websocket.AddSuccessfulSubscriptions(s)
+		}
+		if err != nil {
+			errs = common.AppendError(errs, err)
+		}
 	}
-	b.Websocket.AddSuccessfulSubscriptions(subs...)
-	return nil
+
+	return errs
 }
 
 // Unsubscribe sends a websocket message to manage and remove a subscription.
-func (b *BTCMarkets) Unsubscribe(subs []subscription.Subscription) error {
-	payload := WsSubscribe{
-		MessageType: removeSubscription,
-		ClientType:  clientType,
-	}
-	for i := range subs {
-		payload.Channels = append(payload.Channels, subs[i].Channel)
-		if subs[i].Pair.IsEmpty() {
-			continue
+func (b *BTCMarkets) Unsubscribe(subs subscription.List) error {
+	var errs error
+	for _, s := range subs {
+		req := WsSubscribe{
+			MessageType: removeSubscription,
+			ClientType:  clientType,
+			Channels:    []string{s.Channel},
+			MarketIDs:   s.Pairs.Strings(),
 		}
 
-		pair := subs[i].Pair.String()
-		if common.StringDataCompare(payload.MarketIDs, pair) {
-			continue
+		err := b.Websocket.Conn.SendJSONMessage(req)
+		if err == nil {
+			err = b.Websocket.RemoveSubscriptions(s)
+		}
+		if err != nil {
+			errs = common.AppendError(errs, err)
 		}
-		payload.MarketIDs = append(payload.MarketIDs, pair)
-	}
-
-	err := b.Websocket.Conn.SendJSONMessage(payload)
-	if err != nil {
-		return err
 	}
-	b.Websocket.RemoveSubscriptions(subs...)
-	return nil
+	return errs
 }
 
 // ReSubscribeSpecificOrderbook removes the subscription and the subscribes
 // again to fetch a new snapshot in the event of a de-sync event.
 func (b *BTCMarkets) ReSubscribeSpecificOrderbook(pair currency.Pair) error {
-	sub := []subscription.Subscription{{
+	sub := subscription.List{{
 		Channel: wsOB,
-		Pair:    pair,
+		Pairs:   currency.Pairs{pair},
```

Comment:
This needs testing, since we didn't subscribe to a single MarketID, we might not be able to do a single resub.